examples/master/master.ino

```diff
@@ -2,20 +2,124 @@
 
 bool addresses[127] = {false};
 
-// Addresses pins
-int as[] = {8, 9, 10, 11};
+uint16_t timeleft = 300;   // 5 minutes
+uint16_t difficulty = 1;   // Easy
 
-uint16_t timeleft = 300; // 5 minutes
-uint16_t difficulty = 1;
+bool defused = false;      // Is the bombe defused ?
 
-void initAddresses()
+// Message's structure
+struct BusMessage
 {
-  for (int i = 0; i < sizeof(as) / sizeof(as[0]); i++) {
-    pinMode(as[i], OUTPUT);
-    digitalWrite(as[i], HIGH);
+  uint16_t data[2];
+};
+
+// Output
+#define RED_LED (2)
+#define GREEN_LED (3)
+#define YELLOW_LED (4)
+
+// Input
+#define SENSE_PIN (0)
+
+// Command
+# define TIME_CMD (0)
+# define DIFFICULTY_CMD (1)
+# define NEED_TO_SPEAK_CMD (2)
+# define INFO_CMD (3)
+
+# define DEFUSED_CMD (4)
+# define PENALITY_CMD (5)
+# define END_CMD (6)
+
+// Answer
+# define ANSWER_YES (1)
+# define ANSWER_NO (0)
+# define ANSWER_OK (1)
+
+void printDeviceFound(uint8_t address)
+{
+  Serial.print("I2C device found at address 0x");
+  if (address < 16)
+    Serial.print("0");
+  Serial.println(address, HEX);
+}
+
+void printUnknownError(uint8_t address)
+{
+  Serial.print("Unknown error at address 0x");
+  if (address < 16)
+    Serial.print("0");
+  Serial.println(address, HEX);
+}
+
+void printNumberOfDeviceFound(byte n)
+{
+  if (n == 0)
+    Serial.println("No I2C devices found\n");
+  else
+  {
+    Serial.print("Found ");
+    Serial.print(n);
+    Serial.println(" device(s).");
   }
 }
 
+void printHeader(int command, int addr, byte res)
+{
+  Serial.print("=== command : ");
+  Serial.print(command);
+  Serial.print(" | addr : ");
+  Serial.print(addr);
+  Serial.print(" | response : ");
+  Serial.print(res);
+  Serial.print(" | timeleft : ");
+  Serial.print(timeleft);    
+  Serial.println(" ===");
+}
+
+void writeUint16(uint16_t i)
+{
+  Wire.write(i);
+  Wire.write(i >> 8);
+}
+
+BusMessage sendTo(int addr, uint16_t command, uint16_t parameter)
+{
+  Wire.beginTransmission(addr);
+  writeUint16(command);
+  writeUint16(parameter);
+  Wire.endTransmission();
+  delay(1);
+  return receiveFrom(addr);
+}
+
+BusMessage sendTime(int addr)
```

Comment:
The following functions are useless and memory consuming.
`sendDifficulty(42) is as explicit as ask(DIFFICULTY, difficulty, 42)`